core/java/iesi-core/src/main/java/io/metadew/iesi/openapi/ComponentParser.java

```diff
@@ -0,0 +1,290 @@
+package io.metadew.iesi.openapi;
+
+import io.metadew.iesi.connection.database.oracle.OracleDatabaseService;
+import io.metadew.iesi.metadata.definition.component.Component;
+import io.metadew.iesi.metadata.definition.component.ComponentParameter;
+import io.metadew.iesi.metadata.definition.component.ComponentVersion;
+import io.metadew.iesi.metadata.definition.component.key.ComponentKey;
+import io.metadew.iesi.metadata.definition.component.key.ComponentParameterKey;
+import io.metadew.iesi.metadata.definition.component.key.ComponentVersionKey;
+import io.swagger.v3.oas.models.OpenAPI;
+import io.swagger.v3.oas.models.Operation;
+import io.swagger.v3.oas.models.PathItem;
+import io.swagger.v3.oas.models.Paths;
+import io.swagger.v3.oas.models.parameters.Parameter;
+import io.swagger.v3.oas.models.parameters.RequestBody;
+import io.swagger.v3.oas.models.responses.ApiResponses;
+import io.swagger.v3.oas.models.security.SecurityRequirement;
+import io.swagger.v3.oas.models.security.SecurityScheme;
+
+import java.util.*;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class ComponentParser {
+    private static ComponentParser INSTANCE;
+
+    private static Map<String, SecurityScheme> securitySchemeMap;
+    private static Long versionNumber;
+    private static String connectionName;
+
+
+    private ComponentParser() {}
+
+
+    public synchronized static ComponentParser getInstance() {
+        if (INSTANCE == null) {
+            INSTANCE = new ComponentParser();
+        }
+        return INSTANCE;
+    }
+
+    public List<Component> parse(OpenAPI openAPI) {
+        List<Component> components = new ArrayList<>();
+        securitySchemeMap = openAPI.getComponents().getSecuritySchemes();
+        versionNumber = Long.parseLong(openAPI.getInfo().getVersion());
+        connectionName = openAPI.getInfo().getTitle();
+
+
+
+        Paths paths = openAPI.getPaths();
+
+        for (String pathName : paths.keySet()) {
+            PathItem path = paths.get(pathName);
+            Map<PathItem.HttpMethod, Operation> operations = path.readOperationsMap();
+
+            for (PathItem.HttpMethod httpMethod : operations.keySet()) {
+                components.addAll(initComponents(operations.get(httpMethod), httpMethod, pathName));
+            }
+
+        }
+
+       return components;
+    }
+
+    private  List<Component> initComponents(Operation operation, PathItem.HttpMethod operationName, String pathName) {
+        List<Component> components = new ArrayList<>();
+        if (operation != null) {
+
+            Set<String> securities = getSecurities(operation.getParameters(), operation.getSecurity());
```

Comment:
cant you just provide the operation to the method and retrieve the parameters and security in the method?